test/deal-investor/post-deal-investors.api-test.ts

```diff
@@ -0,0 +1,309 @@
+    makeRequest: (body) => api.post(createDealInvestorUrl, body),
+    givenAnyRequestBodyWouldSucceed: () => {
+      givenAuthenticationWithTheIdpSucceeds();
+      givenAnyRequestBodyToCreateDealInvestorInAcbsSucceeds();
+    },
+  });
+
+  withStringFieldValidationApiTests({
+    fieldName: 'currency',
+    minLength: 0,
+    maxLength: 3,
+    generateFieldValueOfLength: (length: number) => valueGenerator.string({ length }),
+    validRequestBody: requestBodyToCreateDealInvestor,
+    makeRequest: (body) => api.post(createDealInvestorUrl, body),
+    givenAnyRequestBodyWouldSucceed: () => {
+      givenAuthenticationWithTheIdpSucceeds();
+      givenAnyRequestBodyToCreateDealInvestorInAcbsSucceeds();
+    },
+  });
+
+  it('returns a 404 response if ACBS responds with a 400 response that is a string containing "The deal not found"', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToCreateDealInvestor().reply(400, 'The deal not found or user does not have access');
+
+    const { status, body } = await api.post(createDealInvestorUrl, requestBodyToCreateDealInvestor);
+
+    expect(status).toBe(404);
+    expect(body).toStrictEqual({ message: 'Not found', statusCode: 404 });
+  });
+
+  it('returns a 400 response if ACBS responds with a 400 response that is not a string', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    const acbsErrorMessage = { Message: 'error message' };
+    requestToCreateDealInvestor().reply(400, acbsErrorMessage);
+
+    const { status, body } = await api.post(createDealInvestorUrl, requestBodyToCreateDealInvestor);
+
+    expect(status).toBe(400);
+    expect(body).toStrictEqual({ message: 'Bad request', error: JSON.stringify(acbsErrorMessage), statusCode: 400 });
+  });
+
+  it('returns a 400 response if ACBS responds with a 400 response that is a string that does not contain "The deal not found"', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    const acbsErrorMessage = 'ACBS error message';
+    requestToCreateDealInvestor().reply(400, acbsErrorMessage);
+
+    const { status, body } = await api.post(createDealInvestorUrl, requestBodyToCreateDealInvestor);
+
+    expect(status).toBe(400);
+    expect(body).toStrictEqual({ message: 'Bad request', error: acbsErrorMessage, statusCode: 400 });
+  });
+
+  it('returns a 500 response if ACBS responds with an error code that is not 400"', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToCreateDealInvestor().reply(401, 'Unauthorized');
+
+    const { status, body } = await api.post(createDealInvestorUrl, requestBodyToCreateDealInvestor);
+
+    expect(status).toBe(500);
+    expect(body).toStrictEqual({ message: 'Internal server error', statusCode: 500 });
+  });
+
+  const givenRequestToCreateDealInvestorInAcbsSucceeds = (): nock.Scope => {
+    return requestToCreateDealInvestor().reply(201);
+  };
+
+  // TO-DO: do we need to test timeout for ACBS /POST?
```

Comment:
Do we need to add a test for the /POST to ACBS timing out? There didn't appear to be this test for some of the latest post endpoints (e.g. deal guarantee)

Reply:
I guess we don't handle timeouts in each endpoint separately and we don't repeatedly test it?

@lmarrai-sw ?

Reply:
Good question, I didn't deliberately stop including the test but I did forget to include it in my last few tickets
I think we should have it

Reply:
Added